apps/demo/src/h5wasm/UrlForm.tsx

```diff
@@ -0,0 +1,86 @@
+import useAxios from 'axios-hooks';
+import type { FormEvent } from 'react';
+import { useCallback } from 'react';
+import { useEffect } from 'react';
+import { FiLoader } from 'react-icons/fi';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import styles from './H5WasmApp.module.css';
+import type { H5File } from './models';
+
+interface Props {
+  onH5File: (h5File: H5File) => void;
+}
+
+function UrlForm(props: Props) {
+  const { onH5File } = props;
+
+  const navigate = useNavigate();
+  const query = new URLSearchParams(useLocation().search);
+  const url = query.get('url') || '';
+
+  const [{ loading, error }, execute] = useAxios<ArrayBuffer>(
+    { url, responseType: 'arraybuffer' },
+    { manual: true }
+  );
+
+  const fetchFile = useCallback(async () => {
+    if (url) {
+      const { data } = await execute();
+      onH5File({ filename: url.slice(url.lastIndexOf('/') + 1), buffer: data });
+    }
+  }, [url, execute, onH5File]);
+
+  useEffect(() => {
+    void fetchFile();
+  }, [url, fetchFile]);
+
+  function handleUrlSubmit(evt: FormEvent<HTMLFormElement>) {
+    evt.preventDefault();
+    const formData = new FormData(evt.currentTarget);
+    const newUrl = formData.get('url') as string;
+
+    if (newUrl === url) {
+      void fetchFile(); // refetch
```

Comment:
When an error occurs and the users submits the form again, navigating to the same URL would not re-invoke the `useEffect`, so I refetch manually instead.